### ISSUE TITLE
error-nil: add Empty and Zero

### ISSUE THREAD

**Issue by ccoVeille** [Antonboom/testifylint#138]
we do not detect error used with empty/zero, right now

```go
var err error
assert.Empty(t, err)
assert.Zero(t, err)
```

We should recommend using `NoError`

```go
var err error
assert.NoError(t, err)
```

this pattern is widely used https://github.com/search?q=language%3Ago+%22assert.Empty%28t%2C+err%22&type=code
https://github.com/search?q=language%3Ago+%22assert.Zero%28t%2C+err%29%22&type=code

These two are candidates for `error-nil`

_Originally posted by @ccoVeille in https://github.com/Antonboom/testifylint/issues/119#issuecomment-2181588845_

**Comment by Antonboom** [comment]
Yes, this is valid for me.

Need to add to ignored cases in tests:
```go
assert.Empty(t, errs)
assert.Empty(t, err.String())
```
